### PR TITLE
Improve error messages for loading tls keys

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -1463,7 +1463,10 @@ func lookupGidByName(nameOrGid string) (int, error) {
 func setupTls(cert, key, ca string, l net.Listener) (net.Listener, error) {
 	tlsCert, err := tls.LoadX509KeyPair(cert, key)
 	if err != nil {
-		return nil, fmt.Errorf("Couldn't load X509 key pair (%s, %s): %s. Key encrypted?",
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("Couldn't load X509 key pair (%s, %s): %v", cert, key, err)
```
Comment (tiborvass) [inline api/server/server.go]: while you're at it let's remove the contraction: should be `Could not`.

**Comment by duglin** [comment]
Perhaps a testcase to test the various error flows?

**Comment by thaJeztah** [comment]
This PR was recently merged; https://github.com/docker/docker/pull/11073/files should that be updated the same way?

**Comment by cpuguy83** [comment]
@duglin I don't see the purpose of having a test to ensure that the friendlier error message is displayed here.

**Comment by duglin** [comment]
we like tests?  :-)
No biggie - just habit.

**Comment by tiborvass** [comment]
LGTM

**Comment by duglin** [comment]
LGTM

**Pull request merged**